### PR TITLE
2020 override commands

### PR DESCRIPTION
**Pull request by Pixlskull**
Replaced OverrideCommandGroup with OverrideParallelCommandGroup and OverrideSequentialCommandGroup, and updated OverrideSet

### CODE REVIEW

File: commands/OverridableSequentialCommandGroup.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	protected final Overridable overridable;
+
+	public OverridableSequentialCommandGroup(Overridable overridable) {
+		super();
+		this.overridable = overridable;
+	}
+
+	public final synchronized void addSequentialUnlessOverridden(Command command) {
+        addCommands(new RunIf(command, overridable::isNotOverridden));
+		//addSequential(new RunIf(command, overridable::isNotOverridden));
```
Comment (aemino) [inline commands/OverridableSequentialCommandGroup.java]: why is this commented out?

File: commands/solenoid/SolenoidSet.java [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
 import org.usfirst.frc4904.standard.subsystems.SolenoidSubsystem.SolenoidState;
-import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
 
 /**
  * Command to set the state of a SolenoidSubsystem
  */
-public class SolenoidSet extends Command {
+public class SolenoidSet implements Command {
```
Comment (aemino) [inline commands/solenoid/SolenoidSet.java]: I think this needs to extend `CustomCommandBase`.

File: commands/solenoid/SolenoidSet.java [hunk 3]
```diff
@@ -23,34 +26,21 @@
...
 
-	/**
-	 * Sets the state of a SolenoidSubsystem
-	 * 
-	 * @param system SolenoidSubsystem to set
-	 * @param state  state to set system
-	 */
-	public SolenoidSet(SolenoidSubsystem system, SolenoidState state, BooleanSupplier... booleanSuppliers) {
-		this("SolenoidSet", system, state, booleanSuppliers);
+		requirements = Set.of((Subsystem) system);
```
Comment (aemino) [inline commands/solenoid/SolenoidSet.java]: Don't add this property. Requirements should be computed solely in `getRequirements`

File: commands/solenoid/SolenoidSet.java [hunk 4]
```diff
@@ -64,11 +54,19 @@ public void initialize() {
 		system.set(state);
 	}
 
+	/**
+	 * Gets requirements
+	 */
+	public Set<Subsystem> getRequirements() {
+		return requirements;
```
Comment (aemino) [inline commands/solenoid/SolenoidSet.java]: The `Set.of` call should go here

File: subsystems/SolenoidSubsystem.java [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
 import edu.wpi.first.wpilibj.DoubleSolenoid;
-import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj2.command.Subsystem;
 
 /**
  * A class that wraps multiple DoubleSolenoid objects with subsystem functionality.
  * Allows for easy inversion and setting of default state of solenoids
  */
-public class SolenoidSubsystem extends Subsystem {
+public class SolenoidSubsystem implements Subsystem {
```
Comment (aemino) [inline subsystems/SolenoidSubsystem.java]: I think we're going to have a `CustomSubsystemBase` class

File: subsystems/SolenoidSubsystem.java [hunk 6]
```diff
@@ -112,8 +64,8 @@ public SolenoidSubsystem(boolean isInverted, DoubleSolenoid... solenoids) {
 	 * @param solenoids
 	 *                  Double solenoids of the system
 	 */
-	public SolenoidSubsystem(DoubleSolenoid... solenoids) {
-		this("SolenoidSubsystem", solenoids);
+	public SolenoidSubsystem( DoubleSolenoid... solenoids) {
```
Comment (aemino) [inline subsystems/SolenoidSubsystem.java]: format: remove extra space

Comment (aemino) [reply subsystems/SolenoidSubsystem.java]: I think this issue is present in another PR too... we should be careful not to merge the same changes multiple times.

File: commands/OverridableSequentialCommandGroup.java (hunk 1 above)
Comment (NikhilSuresh24) [reply commands/OverridableSequentialCommandGroup.java]: just delete this line

File: commands/solenoid/SolenoidExtend.java [hunk 7]
```diff
@@ -14,34 +14,12 @@
 	 * Command to set the state of a SolenoidSubsystem to
 	 * EXTEND(DoubleSolenoid.Value.kForward)
 	 * 
-	 * @param name   Name of Command
```
Comment (NikhilSuresh24) [inline commands/solenoid/SolenoidExtend.java]: don't modify solenoid commands in this branch

File: commands/solenoid/SolenoidOff.java [hunk 8]
```diff
@@ -10,19 +10,6 @@
  * OFF(DoubleSolenoid.Value.kOff)
  */
 public class SolenoidOff extends SolenoidSet {
-	/**
```
Comment (NikhilSuresh24) [inline commands/solenoid/SolenoidOff.java]: same thing above

File: commands/solenoid/SolenoidSet.java (hunk 4 above)
Comment (bcho04) [reply commands/solenoid/SolenoidSet.java]: see #245.

File: commands/solenoid/SolenoidSet.java (hunk 3 above)
Comment (bcho04) [reply commands/solenoid/SolenoidSet.java]: see #245.

File: commands/solenoid/SolenoidSet.java (hunk 2 above)
Comment (bcho04) [reply commands/solenoid/SolenoidSet.java]: see #245.

File: subsystems/SolenoidSubsystem.java (hunk 6 above)
Comment (bcho04) [reply subsystems/SolenoidSubsystem.java]: see #245.

File: commands/solenoid/SolenoidOff.java (hunk 8 above)
Comment (bcho04) [reply commands/solenoid/SolenoidOff.java]: see #245.

File: commands/solenoid/SolenoidExtend.java (hunk 7 above)
Comment (bcho04) [reply commands/solenoid/SolenoidExtend.java]: see #245.

**Comment by bcho04** [comment]
Reverted all Solenoid changes from previous commits using `git revert`.

File: commands/OverridableSequentialCommandGroup.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+public abstract class OverridableSequentialCommandGroup extends SequentialCommandGroup {
+	protected final Overridable overridable;
+
+	public OverridableSequentialCommandGroup(Overridable overridable) {
+		super();
+		this.overridable = overridable;
+	}
+
+	public final synchronized void addSequentialUnlessOverridden(Command command) {
+        addCommands(new RunIf(command, overridable::isNotOverridden));
```
Comment (aemino) [inline commands/OverridableSequentialCommandGroup.java]: indentation in this file is a bit weird, run formatter

File: commands/OverridableParallelCommandGroup.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+
+public abstract class OverridableParallelCommandGroup extends ParallelCommandGroup {
+	protected final Overridable overridable;
+
+	public OverridableParallelCommandGroup(Overridable overridable) {
+		super();
+		this.overridable = overridable;
+	}
+
```
Comment (NikhilSuresh24) [inline commands/OverridableParallelCommandGroup.java]: add a constructor to take in a name and call `setName(name)`

Comment (NikhilSuresh24) [reply commands/OverridableParallelCommandGroup.java]: also make this constructor use that one

**Pull request merged**